pkg/runtime/reconciler_test.go

```diff
@@ -71,9 +73,265 @@ func TestReconcilerUpdate(t *testing.T) {
 	ids := &ackmocks.AWSResourceIdentifiers{}
 	ids.On("ARN").Return(&arn)
 
+	latestRTObj := &k8srtmocks.Object{}
+	latestRTObj.On("GetObjectKind").Return(objKind)
+	latestRTObj.On("DeepCopyObject").Return(latestRTObj)
+
+	latestMetaObj := &metav1mocks.Object{}
+	latestMetaObj.On("GetAnnotations").Return(
+		map[string]string{},
+	)
+	latestMetaObj.On("GetNamespace").Return("default")
+	latestMetaObj.On("GetName").Return("mybook")
+	latestMetaObj.On("GetGeneration").Return(int64(1))
+
+	latest := &ackmocks.AWSResource{}
+	latest.On("Identifiers").Return(ids)
+	latest.On("Conditions").Return([]*ackv1alpha1.Condition{})
+	latest.On("MetaObject").Return(latestMetaObj)
+	latest.On("RuntimeObject").Return(latestRTObj)
+
+	rd := &ackmocks.AWSResourceDescriptor{}
+	rd.On("GroupKind").Return(
+		&metav1.GroupKind{
+			Group: "bookstore.services.k8s.aws",
+			Kind:  "fakeBook",
+		},
+	)
+	rd.On("EmptyRuntimeObject").Return(
+		&fakeBook{},
+	)
+	rd.On("Delta", desired, latest).Return(
+		delta,
+	).Once()
+	rd.On("Delta", desired, latest).Return(ackcompare.NewDelta())
+	rd.On("UpdateCRStatus", latest).Return(true, nil)
+	rd.On("IsManaged", desired).Return(true)
+
+	rm := &ackmocks.AWSResourceManager{}
+	rm.On("ReadOne", ctx, desired).Return(
+		latest, nil,
+	)
+	rm.On("Update", ctx, desired, latest, delta).Return(
+		latest, nil,
+	)
+
+	rmf := &ackmocks.AWSResourceManagerFactory{}
+	rmf.On("ResourceDescriptor").Return(rd)
+
+	reg := ackrt.NewRegistry()
+	reg.RegisterResourceManagerFactory(rmf)
+
+	zapOptions := ctrlrtzap.Options{
+		Development: true,
+		Level:       zapcore.InfoLevel,
+	}
+	fakeLogger := ctrlrtzap.New(ctrlrtzap.UseFlagOptions(&zapOptions))
+	cfg := ackcfg.Config{}
+	metrics := ackmetrics.NewMetrics("bookstore")
+
+	sc := &ackmocks.ServiceController{}
+	kc := &ctrlrtclientmock.Client{}
+	statusWriter := &ctrlrtclientmock.StatusWriter{}
+
+	kc.On("Status").Return(statusWriter)
+	statusWriter.On("Patch", ctx, latestRTObj, client.MergeFrom(desiredRTObj)).Return(nil)
+	kc.On("Patch", ctx, latestRTObj, client.MergeFrom(desiredRTObj)).Return(nil)
+
+	// TODO(jaypipes): Place the above setup into helper functions that can be
+	// re-used by future unit tests of the reconciler code paths.
+
+	// With the above mocks and below assertions, we check that if we got a
+	// non-error return from `AWSResourceManager.ReadOne()` and the
+	// `AWSResourceDescriptor.Delta()` returned a non-empty Delta, that we end
+	// up calling the AWSResourceManager.Update() call in the Reconciler.Sync()
+	// method,
+	r := ackrt.NewReconcilerWithClient(sc, kc, rmf, fakeLogger, cfg, metrics)
```

Comment:
and here, you'd just do:

```go
r := ackrt.NewReconciler(sc, rmf, fakeLogger, cfg, metrics).WithClient(kc)
```

Reply:
Fair enough. We can work on a separate PR that updates the Reconciler interface to use the builder pattern (for the Kubernetes client, logger, etc)